CIPs/0018.md

```diff
@@ -0,0 +1,53 @@
+# CIP [0018]: Change group epoch score calculation for voter rewards
+
+- Date: 2020-08-30
+- Author: @zviadm
+- Status: DRAFT
+
+## Overview
+
+Currently group epoch score is calculated as an average score of its members epoch score
+for voter rewards. 
+
+Because of this, larger groups with more validators in them have a distinct advantage for 
+voters since their average score per epoch is likely going to be higher since they are able to 
+average out individual bad validator scores with better performing validators.
+
+This seems counter productive since for network health it would be better to incentivize voters
+to vote for smaller groups for more decentralization.
+
+Fix for this can be very simple, we change group epoch score to be:
+* Minimum(member validator epoch scores), instead of Avg(member validator epoch scores).
+
+Calculating score per epoch using minimum incetinvizes voting for smaller groups since they
+are more likely to maintain better epoch scores, and it also puts more pressure on larger groups
+to make sure all their validators maintain high uptime.
+
+This is a pretty simple change, and it only needs to be changed in one place here:
+calculateGroupEpochScore function in Validators.sol
+https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Validators.sol#L418
+
+## Goals
+
+- Shift voter incentive to vote for smaller groups instead of larger groups.
+- Provide more incentive for larger groups to maintain high uptime for all of their validators.
+
+## Proposed Solution
+
+
+Change calculateGroupEpochScore function in Validators.sol to calculate group score using 
+a minimum instead of average.
+https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Validators.sol#L418
+
+## Alternative Solutions
+
+## Risks
```

Comment:
This is an interesting point. I don't see this being an issue in practice since most of the large groups are fully self funded/self-voted anyways so for them it probably wont make a big difference. 

Also, as long as all groups are named and verified, it will be questionable to see multiple groups with the same name/domain, and they will go in questionable group category that people should most likely not delegate their votes to.

Reply:
Zviad,

This makes sense and should help to level the playing field for validators.  Supportive.

Reply:
I also support this.

Reply:
@asaj and I considered several metrics here when putting this together initially. ThIe choice of AVG was because it is means validator and groups are invariant without requiring any Sybil resistance -- i.e. as @nategraf says, there is no advantage to be gained by groups splitting themselves into individual units. If we made this change, the rational behavior for any multi-validator groups would be to split into singletons, and I think that would potentially confuse users and diminish the value of groups. FWIW I share the sentiment and considered several other ways of disincentivizing forming of larger groups, like a diminishing reward curve, but without Sybil resistance all are problematic.

Reply:
I understand worries about Sybil attacks, but I think there are already enough incentives that will deter groups from doing it. Also 5 validator per group limit would also be pointless if it was that simple to just spin up separate groups for the same identity. 

I think reputation/community risk is far too high for a group to do something that we would consider a bad practice, this is assuming group actually depends on others to vote for them. For groups that are completely self funded and don't care about anyone else voting for them, the difference between running single group vs multiple is going to be very marginal anyways and they wouldnt care what their "returns" or average score looks like either.

Reply:
Isn't it a desirable property that groups would have to split up into individual units? I.e. at the margins, we do want voters to vote for groups that are more stable and a group that splits itself up is less likely to be voted for? Put differently, all things being equal, is a group less likely to receive votes when split up vs. when consolidated?

Reply:
Given the behavior we have observed so far, it seems likely to me that larger (e.g. 5 member) groups would remain whole due to the operational and reputational costs of splitting. Self-funded groups may split, but in the end it's very difficult to impose any incentive structure on self-funded groups outside their large stake in the price of CELO.

I might expect self-funded groups, especially the ones who have chosen not to identify (of which there are currently 4 groups running a total of 11 validators) to split following this proposal. As a result they will effectively continue to obtain rewards over the average of their scores (specifically the weighted average by how many votes they put into each singleton group). This would put them at a relative advantage to the other larger groups who will then be obtaining the minimum of their scores, and could potentially result in the further accumulation of funds by whales. This effect may be counteracted by the more general spread of CELO tokens to new holders who's votes will dilute the power of whales and reduce the number of validators they can keep elected on their own. I am not sure which of the two of these effects would win out.

Reply:
To keep history, pasting comments from discord here too. 

First, there is already a much larger incentive that exists to split up group into smaller groups as a Sybil attack. This is the slashing penalty. Since slashing penalty is applied to the whole group even if one validator is slashed, it is much safer to run multiple groups instead of one large one. But as we observe, no group is really doing this. Of course, once slashing actually starts working, we will most likely see self-funded groups do all sorts of potentially "not intended" things if they were to get slashed, but there isn't much we can do to control fully self funded groups anyways. 

So if we are looking at not fully self funded groups, there are already many incentives to discourage any "non approved" behaviour like setting up multiple groups:
* Reputation risk
* 2x more CELO locked for 6 months.
* Much harder vote management. It is a lot harder to get other users to vote in perfect balance for multiple groups, especially if owner is hiding the fact that they all belong to the same entity.
* Pretty limited overall benefit to doing this. Since group epoch score isn't cumulative, there really isn't a huge difference in rewards due to occasional outages. If it is calculated as average or minimum it really won't make a difference to go through all these trouble.

Reply:
> * Reputation risk

I believe most of the whales don't depend much on others to vote for them, so the reputation risk is not a factor. 

> * 2x more CELO locked for 6 months.

The amount of LockedGold is the same in whichever arrangement (since the amount you need to lock at a group scales with the number of members).

> * Much harder vote management. It is a lot harder to get other users to vote in perfect balance for multiple groups, especially if owner is hiding the fact that they all belong to the same entity.

For the same reason as described above, where whales don't rely on others' votes and can split their funds into multiple accounts, this isn't much of a hurdle.

> * Pretty limited overall benefit to doing this. Since group epoch score isn't cumulative, there really isn't a huge difference in rewards due to occasional outages. If it is calculated as average or minimum it really won't make a difference to go through all these trouble.

If the rewards from their own voting funds would suffer if any of their validators went down (the whole point of this proposal) they would have a financial incentive to form singleton groups.  

I am definitely interested in exploring ways we can further incentivize voting to support decentralization and broader participation of more validators. In general, whales can organize themselves exactly how n small validators would, and it's very hard to build a mechanism that gets round that. Leveraging Foundation voting is a good avenue -- since the foundation votes are a form of KYC. You can imagine a number of things that could be tied to the "identity" aspect of being a foundation vote recipient.

Reply:
Regarding the big groups splitting, I think that it's possible although with a reputational risk. Also, it is true that those groups are mostly self-funded today, but as adoption grows the outsider votes are going to get more significant. Then having to ask your voters to split their votes across multiple groups and losing the spotlight in the validator ranking by votes, could make groups to be reluctant to split their group.

And even if they do split their group and votes, they still have an operational overhead that smaller groups don't have.

I am supportive of this proposal.

Reply:
Adding to what @nategraf mentioned, this could make existing groups move their validators into anonymity, which is something that can't be prevented but it's not very desired.

Reply:
> > * 2x more CELO locked for 6 months.
> 
> The amount of LockedGold is the same in whichever arrangement (since the amount you need to lock at a group scales with the number of members).
> 

This is true but if they want to split their groups now they'll have to lock liquid CELO (because the one in the group is locked) and then locke this new one for further 6 months. Spliting does have a cost on this regard but if whales are the one doing it then it's not that much.